internal/tui/model_list.go

```diff
@@ -138,20 +171,73 @@ type errorMsg struct {
 	Err error
 }
 
+type keyMap struct {
+	Save  key.Binding
+	Enter key.Binding
+	Quit  key.Binding
+}
+
+func (k keyMap) ShortHelp() []key.Binding {
+	return []key.Binding{k.Save, k.Enter, k.Quit}
+}
+
+func (k keyMap) FullHelp() [][]key.Binding {
+	return [][]key.Binding{
+		{k.Save, k.Enter, k.Quit},
+	}
+}
+
+var inputKeys = keyMap{
+	Save: key.NewBinding(
+		key.WithKeys("s", "enter"),
+		key.WithHelp("enter/ctrl+s", "save suggest"),
+	),
+	Quit: key.NewBinding(
+		key.WithKeys("esc", "esc", "ctrl+c"),
+		key.WithHelp("esc/ctrl+c", "quit"),
+	),
+}
+
 func (m ListModel) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
+	var cmd tea.Cmd
 	switch msg := msg.(type) {
 	case tea.WindowSizeMsg:
 		h, v := appStyle.GetFrameSize()
 		m.list.SetSize(msg.Width-h, msg.Height-v)
+		m.help.Width = msg.Width - h
 
 	case spinner.TickMsg:
 		var cmd tea.Cmd
 		m.spinner, cmd = m.spinner.Update(msg)
 		return m, cmd
 
 	case tea.KeyMsg:
-		kmap := m.KeyMap()
-		if kmap.Matches(msg, "enter") {
+		if m.input.Focused() {
+			switch msg.Type {
+			case tea.KeyEnter, tea.KeyCtrlS:
+				m.list.SetItem(m.list.Cursor(), createListItem(removeSpecialChars(m.input.Value()), m.usePrefix))
+				m.input.Blur()
+				return m, nil
+			case tea.KeyCtrlC, tea.KeyEsc:
+				m.input.Blur()
+				return m, nil
+			}
+
+			m.input, cmd = m.input.Update(msg)
+			return m, cmd
+		}
+		switch msg.String() {
+
+		case "up":
+			m.list.CursorUp()
+
+			return m, nil
+		case "down":
+			m.list.CursorDown()
+
+			return m, nil
+		case "enter":
+
```

Comment:
```suggestion
		kmap := m.KeyMap()

		if kmap.Matches(msg, "t") {
			m := m.TogglePrefixes()
			return m, nil
		}

		if kmap.Matches(msg, "e") {
			selected := m.list.SelectedItem().(item)
			m.input.SetValue(selected.title)
			m.input.Focus()
			return m, nil
		}

		if kmap.Matches(msg, "enter") {
```

If you migrate to the `KeyMap` system, you'll do this instead